### PR TITLE
resolve: Querify most cstore access methods (subset 2)

### PR DESCRIPTION
**Pull request by petrochenkov**
These changes are less likely to cause perf regressions than the rest of https://github.com/rust-lang/rust/pull/108346.

### CODE REVIEW

File: compiler/rustc_resolve/src/build_reduced_graph.rs [hunk 1]
```diff
@@ -115,34 +114,25 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
...
-                        parent,
-                        ModuleKind::Def(def_kind, def_id, name),
-                        expn_id,
-                        self.def_span(def_id),
-                        // FIXME: Account for `#[no_implicit_prelude]` attributes.
-                        parent.map_or(false, |module| module.no_implicit_prelude),
-                    ))
-                }
-                _ => None,
+            let def_kind = self.tcx.def_kind(def_id);
```
Comment (petrochenkov) [inline compiler/rustc_resolve/src/build_reduced_graph.rs:117]: Unintended change slipped through.

**Comment by cjgillot** [comment]
r=me in this form.
(Except if you want to make additional changes before landing it.)

**Pull request merged**